### DISCUSSION TITLE
Themes without JS

### DISCUSSION THREAD

**Discussion by fenugrec** [mkdocs/mkdocs#3155] [Q&A] [answered]
Hi,
I was wondering to what extent the built-in "mkdocs" theme is dependant on javascript ? I understand at least the Search function absolutely requires it. Also the fact that it's built on top of the "bootstrap" framework makes me think maybe nothing would work at all, at least for that particular theme.

If that's the case, does anyone know of a mkdocs theme that would work reasonably well without JS ?

Thanks;

**Comment by kamilkrzyskow** [comment, marked as answer]
I'm not sure what is your expectation of "works reasonably well", but JavaScript only causes issues when the DOM structure of the website is modified by it, like the Search functionality processes data and then dynamically creates a list of clickable buttons filtered to your search prompt, another case are modals/pop-up screens (just like one when you press the 🔍 icon at the top of the [site](https://www.mkdocs.org/)). However, not everything dynamic in a site is run by JavaScript, take for example the [\<details\>](https://www.w3schools.com/TAGS/tag_details.asp) HTML5 tag.

You can check how the website behaves without JavaScript by disabling it in your browser Developer Tools, typically found under the F12 key ([guide](https://developer.chrome.com/docs/devtools/javascript/disable/)). I just tested it on Firefox and it was a little bit buggy (it randomly enabled it back), so I would recommend using Chrome for that.

I'm using the theme with JavaScript, but the [Material](https://squidfunk.github.io/mkdocs-material/) theme does a good job with providing non-JavaScript solutions out of the box.
For example alongside the theme, the [PyMdown Extensions](https://facelessuser.github.io/pymdown-extensions/) module is installed (can be used with other themes), which [adds an option](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/#highlight) to use [Pygments](https://pygments.org/) for code highlighting. This way it's purely HTML and CSS based. 

I didn't use it myself, but I guess the "mkdocs" theme, only allows to use the `Highlight.js` out of the box, which as the name entails, would require JavaScript. So if you want as little friction as possible with setting the page up to work without JavaScript, then try the Material theme first and see if it caters to your needs.

**Reply by fenugrec** [reply]
Thanks, exactly the type of info I was looking for.

My question was a bit vague because my requirements are still somewhat vague, but I was worried that things like the navigation tree / table of contents might be driven by js elements... My limited knowledge of HTML predates HTML5 so I only have a vague idea of what is possible with just HTML5+CSS.

I had initially set aside `Material` as it seemed tightly integrated with js functions, but will definitely take a closer look.

**Reply by kamilkrzyskow** [reply]
Any progress on your research @fenugrec? What theme did you pick? Also if this resolved your issue please consider marking it as the answer.

**Comment by fenugrec** [comment]
Ah yes, I reviewed the docs for Material, decided to give it a try. Ran into some problems installing mkdocs and its many dependencies : only unofficial packages are available on my distro (Arch), not this project's fault. Decided to try hugo with a very rudimentary theme (`hugo-book` , with some minor customizations) for "a quick test", and that quick test proved to be good enough so I will be using that unless I encounter more problems or limitations.
Thanks again for your initial reply which indeed covered the main points I was wondering about.

**Reply by kamilkrzyskow** [reply]
I'm sorry to hear that you had issues installing MkDocs. I'm not into Linux that much, therefore I don't know of any remedies. Maybe Docker could allow to bypass the main OS lack of support. 
When I searched for `Hugo very rudimentary theme` nothing specific with those keywords showed up 😉 
So if your Hugo theme has a specific name, maybe provide it here for others to know. Good luck with your choice. 👋
